backend/fregepoc/repositories/serializers.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework import serializers
+
+from fregepoc.repositories.models import Repository, RepositoryFile
+
+
+class RepositorySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Repository
+        fields = ['id', 'description', 'analyzed', 'git_url', 'repo_url', 'commit_hash', 'discovered_time',
+                  'fetch_time']
+
+
+class RepositoryFileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = RepositoryFile
+        fields = ['id', 'repository', 'analyzed', 'language', 'repo_relative_file_path', 'metrics', 'analyzed_time']
```

Comment:
It might be beneficial to use [drf-flex-fields](https://github.com/rsinger86/drf-flex-fields) to selectively extend the `repository` field.

Reply:
A single repository can contain thousands of files, as I see there is no pagination support in drf-flex-fields - this might cause performance problems on large repositories. Current solution (filters) support pagination out of the box.

Reply:
This is correct (although it's the rest's pagination class that supports pagination, not filters - unless we want to manually paginate via fields), marking as resolved.